src/NexusMods.StandardGameLocators/AGameLocator.cs

```diff
@@ -21,21 +22,22 @@ protected AGameLocator(ILogger logger, AHandler<TRecord, TId> handler)
 
     public IEnumerable<GameLocatorResult> Find(IGame game)
     {
-        if (game is not TGame tg) yield break;
+        if (game is not TGame tg) return Enumerable.Empty<GameLocatorResult>();
 
-        foreach (var id in Ids(tg))
+        if (_cachedGames is null)
         {
-            var found = _handler.FindOneGameById(id, out var errors);
-            if (errors.Any() || found == null)
+            _cachedGames = _handler.FindAllGamesById(out var errors);
+            if (errors.Any())
             {
                 foreach (var error in errors)
-                    _logger.LogError("While looking for {Game}: {Error}", game, error);
-            }
-            else
-            {
-                yield return new GameLocatorResult(Path(found));
+                    _logger.LogError("While looking for games: {Error}", error);
             }
         }
+
+        return Ids(tg)
+            .Where(id => _cachedGames.ContainsKey(id))
+            .Select(id => _cachedGames[id])
```

Comment:
PR LGTM.

Just note the double dictionary lookup, a non-LINQ solution would save you a line [and some perf] here, even if it's not as pleasing to the eye.

```csharp
foreach (var id in Ids(tg))
    if (_cachedGames.TryGetValue(id, out var found))
        yield return new GameLocatorResult(Path(found));
```

Reply:
Yeah, `TryGet` pattern is not very LINQ friendly.